src/OpenTK/Platform/Windows/WinRawJoystick.cs

```diff
@@ -341,16 +341,52 @@ public unsafe bool ProcessEvent(IntPtr raw)
 
         HatPosition GetHatPosition(uint value, HidProtocolValueCaps caps)
         {
+            if (caps.LogicalMax == 3)
+            {
+                //4-way hat switch as per the example in Appendix C
+                //http://www.usb.org/developers/hidpage/Hut1_12v2.pdf
+                switch (value)
+                {
+                    case 0:
+                        return HatPosition.Left;
+                    case 1:
+                        return HatPosition.Up;
+                    case 2:
+                        return HatPosition.Right;
+                    case 3: 
+                        return HatPosition.Down;
+                    default:
+                        return HatPosition.Centered;
+                }
+            }
             if (caps.LogicalMax == 8)
+            {
+                //Hat states are represented as a plain number from 0-8
+                //with centered being zero
+                //Padding should have already been stripped out, so just cast
+                if (value > 8)
+                {
+                    //Value out of bounds, so return centered
+                    return HatPosition.Centered;
+                }
                 return (HatPosition)value;
-            else if (caps.LogicalMax == 7)
+            }
+            if (caps.LogicalMax == 7)
             {
+                //Hat states are represented as a plain number from 0-7
+                //with centered being 8
+                if (value > 8)
+                {
+                    //Return zero if our value is out of bounds ==> e.g.
+                    //Thrustmaster T-Flight Hotas X returns 15 for the centered position
+                    return HatPosition.Centered;
+                }
```

Comment:
Take these two checks and place them at the start of the function instead. The LogicalMax value is the absolute max value that the device should report, and as such we don't need to duplicate the check. 

```cs
if (value > caps.LogicalMax)
{
    return HatPosition.Centered;
}
```
at 344 should suffice. This also eliminates the need for the default clause in the 4-position hat.